### PR TITLE
Removes the simple metric builders in place of AggFactory implementations

### PR DESCRIPTION
**Pull request by colings86**
This follows on from https://github.com/elastic/elasticsearch/pull/15821 and requires that PR is merged first

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilders.java [hunk 1]
```diff
@@ -89,8 +91,8 @@ private AggregationBuilders() {
     /**
      * Create a new {@link ValueCount} aggregation with the given name.
      */
-    public static ValueCountBuilder count(String name) {
-        return new ValueCountBuilder(name);
+    public static ValueCountAggregator.Factory<ValuesSource> count(String name) {
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilders.java]: We should not expose the ValuesSource class in the client API. Maybe this class does not need to be generic at all?

File: core/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilders.java [hunk 2]
```diff
@@ -271,8 +273,8 @@ public static PercentileRanksBuilder percentileRanks(String name) {
     /**
      * Create a new {@link Cardinality} aggregation with the given name.
      */
-    public static CardinalityBuilder cardinality(String name) {
-        return new CardinalityBuilder(name);
+    public static CardinalityAggregatorFactory<ValuesSource> cardinality(String name) {
+        return new CardinalityAggregatorFactory<ValuesSource>(name, ValuesSourceType.ANY, null);
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilders.java]: Similar concerns as ValueCountAggregator.

**Comment by jpountz** [comment]
I left some comments. This change also makes me think that we should move factories to their own classes in all cases since it's weird today to have to import Aggregator impls which are internal to the server side.

**Comment by colings86** [comment]
@jpountz thanks for the review. I think you are right with the generics not being needed so I have removed them from the Cardinality, Missing and ValuesSource Aggregator Factories (which are all the metric factory classes which had been made generic)

I also agree with moving the factories to their own classes and had already intended to do that but I wanted to do it at the end of the refactoring of the aggregations so it makes the diffs easier to review. Does that work for you?

**Comment by jpountz** [comment]
Totally!

**Comment by jpountz** [comment]
LGTM

**Pull request merged**